src/lib.rs

```diff
@@ -97,20 +98,99 @@ pub fn select_coin_bnb(
     inputs: &[OutputGroup],
     options: CoinSelectionOpt,
 ) -> Result<SelectionOutput, SelectionError> {
-    unimplemented!()
+    let mut selected_inputs : Vec<usize> =  vec![] ; 
+    let bnb_tries = 1000000 ; 
+    let bnb_selected_coin = bnb(&desce_input(inputs),&mut selected_inputs , 0 , 0 , bnb_tries , &options);
+    if bnb_selected_coin.is_none() {
+       return  select_coin_srd(inputs, options) ;
+    }else {
+        let selected_coin = bnb_selected_coin.unwrap();
+        let accumulated_value :u64= selected_coin.iter().fold(0, |acc, &i| acc + inputs[i].value);
+        let accumulated_weight : u32  = selected_coin.iter().fold(0, |acc, &i| acc + inputs[i].weight);
+        let estimated_fee = 0 ;
+        let waste = calculate_waste(inputs,&selected_inputs,&options,accumulated_value,accumulated_weight,estimated_fee,);
+        let selection_output =  SelectionOutput {
+             selected_inputs: selected_coin ,
+             waste : WasteMetric(waste) 
+        } ;
+        return Ok(selection_output);
+    }
+   
 }
 
 /// Return empty vec if no solutions are found
+// changing the selected_inputs : &[usize] -> &mut Vec<usize> 
 fn bnb(
     inputs_in_desc_value: &[(usize, OutputGroup)],
-    selected_inputs: &[usize],
-    effective_value: u64,
+    selected_inputs: &mut Vec<usize>,
+    effective_values : u64,
     depth: usize,
     bnp_tries: u32,
-    options: &CoinSelectionOpt,
-) -> Vec<usize> {
-    unimplemented!()
-}
+    options: &CoinSelectionOpt
```

Comment:
get `&mut ThreadRng` as last input parameter in the function argument to avoid creating this object over and over again.